core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -636,7 +622,8 @@ private[spark] class TaskSetManager(
       logInfo("Ignoring task-finished event for " + info.id + " in stage " + taskSet.id +
         " because task " + index + " has already completed successfully")
     }
-    failedExecutors.remove(index)
+
+    blacklistTracker.foreach(_.updateFailureExecutors(stageId, info, Success))
```

Comment:
there is one really annoying detail here.  If there are multiple attempts for one stage (which happens on an executor failure), `info.index` is not referencing the same thing between different `TaskSet`s.  Attempt 2 might only need to run a subset of tasks, in which case index 7 could be partition 58.  I think what you want here is actually `tasks(index).partition`, which will refer to the same logic task across all attempts of a stage.